guides/source/sign_up.md

```diff
@@ -0,0 +1,1652 @@
+```
+
+Then migrate the database:
+
+```bash
+$ rails db:migrate
+```
+
+We can also add validations to require these name fields on every `User`. We'll also want to display the user's full name so let's add a method to combine `first_name` and `last_name` too.
+
+Open `app/models/user.rb` and add the following:
+
+```ruby#7-11
+class User < ApplicationRecord
+  has_secure_password
+  has_many :sessions, dependent: :destroy
+
+  normalizes :email_address, with: ->(e) { e.strip.downcase }
+
+  validates :first_name, :last_name, presence: true
+
+  def full_name
+    "#{first_name} #{last_name}"
+  end
+end
+```
+
+Next, let's add sign up so we can register new users.
+
+### Sign Up Routes & Controller
+
+We now have all the necessary columns in our database to register new users. The next step is to create a route for sign up and the matching controller.
+
+In `config/routes.rb`, let's add a resource for sign up:
+
+```ruby#3
+resource :session
+resources :passwords, param: :token
+resource :sign_up
+```
+
+We're using a singular resource here because we want a singular route for `/sign_up`.
+
+This route directs requests to `app/controllers/sign_ups_controller.rb` so let's create that now.
+
+```ruby
+class SignUpsController < ApplicationController
+  def show
+    @user = User.new
+  end
+end
+```
+
+We're using the `show` action to create a new `User` instance to display the sign up form.
+
+Let's create the form next at `app/views/sign_ups/show.html.erb`:
+
+```erb
+<h1>Sign Up</h1>
+
+<%= form_with model: @user, url: sign_up_path do |form| %>
+  <% if form.object.errors.any? %>
+    <div>Error: <%= form.object.errors.full_messages.first %></div>
+  <% end %>
+
+  <div>
+    <%= form.label :first_name %>
+    <%= form.text_field :first_name, required: true, autofocus: true, autocomplete: "given-name" %>
+  </div>
+
+  <div>
+    <%= form.label :last_name %>
+    <%= form.text_field :last_name, required: true, autocomplete: "family-name" %>
+  </div>
+
+  <div>
+    <%= form.label :email_address %>
+    <%= form.email_field :email_address, required: true, autocomplete: "email" %>
+  </div>
+
+  <div>
+    <%= form.label :password %>
+    <%= form.password_field :password, required: true, autocomplete: "new-password" %>
+  </div>
+
+  <div>
+    <%= form.label :password_confirmation %>
+    <%= form.password_field :password_confirmation, required: true, autocomplete: "new-password" %>
+  </div>
+
+  <div>
+    <%= form.submit "Sign up" %>
+  </div>
+<% end %>
+```
+
+This form collects the user's name, email, and password. We're using the `autocomplete` attribute to suggest to the browser the content requested for these fields.
+
+You'll also notice we set `url: sign_up_path` in the form alongside `model: @user`. Without the `url:` argument, `form_with` would see we have a `User` and send the form to `/users`. Since we want the form to submit to `/sign_up`, the `url:` argument tells the form exactly where to submit the data.
```

Comment:
```suggestion
You'll also notice we set `url: sign_up_path` in the form alongside `model: @user`. Without this `url:` argument, `form_with` would see we have a `User` and send the form to `/users` by default. Since we want the form to submit to `/sign_up`, we set the `url:` to override the default route.
```